src/main/java/graphql/schema/idl/TypeDefinitionRegistry.java

```diff
@@ -623,9 +623,17 @@ public boolean isPossibleType(Type abstractType, Type possibleType) {
             return false;
         } else {
             InterfaceTypeDefinition iFace = (InterfaceTypeDefinition) abstractTypeDef;
-            List<ImplementingTypeDefinition> implementingTypeDefinitions = getAllImplementationsOf(iFace);
-            return implementingTypeDefinitions.stream()
-                    .anyMatch(od -> od.getName().equals(targetObjectTypeDef.getName()));
+
+            return types.values().stream()
+                    .filter(ImplementingTypeDefinition.class::isInstance)
+                    .filter(t -> t.getName().equals(targetObjectTypeDef.getName()))
+                    .map(td -> (ImplementingTypeDefinition<?>) td)
+                    .anyMatch(itd -> itd.getImplements()
+                            .stream()
+                            .map(TypeInfo::getTypeName)
+                            .map(tn -> types.get(tn.getName()))
```

Comment:
We get the `TypeDefinition`, just to keep the existing semantics of that method that verified the type actually exists in the schema.

Reply:
Great, I avoid them too but I had seen others so tried to match general style 😸 will remover.

Reply:
We have evolved the code base BTW - older code still uses`.stream()` but mostly we try to unwind them now to get those extra 0.1% boosts